modules/cas_disk/exp_obj.c

```diff
@@ -530,36 +530,29 @@ int casdsk_exp_obj_create(struct casdsk_disk *dsk, const char *dev_name,
 	exp_obj->owner = owner;
 	exp_obj->ops = ops;
 
-	gd = alloc_disk(1);
-	if (!gd) {
-		result = -ENOMEM;
+	result = _casdsk_init_tag_set(dsk, &dsk->tag_set);
+	if (result) {
+		goto error_init_tag_set;
```

Comment:
Since the module ref count has been incremented a few lines above, you need to call `module_put()` in rollback